### PR TITLE
Add close icon to tabs in Tabs account preferences page

### PR DESCRIPTION
**Pull request by ariasuni**
<details><summary>GIF of the result</summary>

screen_record
</details>

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt [hunk 1]
```diff
@@ -67,17 +69,30 @@ class TabAdapter(private var data: List<TabData>,
...
             } else {
                 false
             }
         }
+        holder.itemView.removeButton?.setOnClickListener {
+            listener.onTabRemoved(holder.adapterPosition)
+        }
+        if (holder.itemView.removeButton != null) {
+            removeButtonEnabled?.let {
+                holder.itemView.removeButton.setEnabled(it)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt]:
```suggestion
                holder.itemView.removeButton.isEnabled = it
```

File: app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt [hunk 2]
```diff
@@ -67,17 +69,30 @@ class TabAdapter(private var data: List<TabData>,
...
+                ThemeUtils.setDrawableTint(
+                    holder.itemView.context,
+                    holder.itemView.removeButton.getDrawable(),
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt]:
```suggestion
                    holder.itemView.removeButton.drawable,
```

File: app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt [hunk 3]
```diff
@@ -32,14 +32,16 @@ import kotlinx.android.synthetic.main.item_tab_preference.view.*
...
     fun onStartDelete(viewHolder: RecyclerView.ViewHolder)
     fun onStartDrag(viewHolder: RecyclerView.ViewHolder)
     fun onActionChipClicked(tab: TabData)
 }
 
 class TabAdapter(private var data: List<TabData>,
                  private val small: Boolean = false,
-                 private val listener: ItemInteractionListener? = null) : RecyclerView.Adapter<TabAdapter.ViewHolder>() {
+                 private val listener: ItemInteractionListener,
+                 private var removeButtonEnabled: Boolean? = null) : RecyclerView.Adapter<TabAdapter.ViewHolder>() {
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/TabAdapter.kt]: Better make `removeButtonEnabled` non-nullable (so its a primitive on the jvm and not an object, also no need for the nullcheck)

**Comment by ariasuni** [comment]
I also removed the default value of `small` since the class is only instanced two times with positional arguments.

**Pull request merged**